src/Confluent.Kafka/Impl/SafeKafkaHandle.cs

```diff
@@ -284,13 +286,13 @@ internal SafeTopicHandle NewTopic(string topic, IntPtr config)
             return topicHandle;
         }
 
-        private IntPtr marshalHeaders(IEnumerable<IHeader> headers)
+        private IntPtr marshalHeaders(IReadOnlyCollection<IHeader> headers)
         {
             var headersPtr = IntPtr.Zero;
 
-            if (headers != null && headers.Any())
+            if (headers != null && headers.Count > 0)
```

Comment:
`Any` may still be better here

Reply:
but not from performance perspective. Any allocates HeadersEnumerator + MoveNext, when Count is just access to the List field.